### DISCUSSION TITLE
Detox - network request timesout errors causing tests to fail

### DISCUSSION THREAD

**Discussion by vigneshs-sportsbet** [wix/Detox#4490] [Q&A]
Tests fails on both Android emulator and iOS Simulator due to below error. Also similar threads regarding this were unanswered. 
Could someone help on this?

```
11:23:10.436 detox[28784] i The app has not responded to the network requests below:
11:23:21    (id = 58) currentStatus: {}
11:23:21    (id = -1000) isReady: {}
```
[3870](https://github.com/wix/Detox/discussions/3870)
[3460](https://github.com/wix/Detox/discussions/3460)

**Comment by portyu9** [comment]
Hi @vigneshs-sportsbet 

These entries:

```text
currentStatus: {}
isReady: {}
```

are Detox's own control/protocol requests. They are not application URLs that should be added to the URL blacklist.

So I would not try to solve this by blacklisting `currentStatus` or `isReady`.

A timeout there generally means the Detox test process is waiting for the app-side Detox client and the app has stopped responding, disconnected, crashed, or become stuck before completing the request.

I would check:

1. Run the failing test with Detox trace/debug logging.
2. Inspect native device/simulator logs for a crash or fatal exception.
3. Check whether startup is stuck on long-running JS/native work.
4. Use Detox synchronization debugging to identify an actual busy resource.
5. Only use URL blacklisting if the diagnostics identify a real long-polling/network endpoint that never becomes idle.

So the `currentStatus` / `isReady` timeout is normally the **symptom**, not the network request that needs blacklisting.

Please mark this answer as accepted if it helped, thank you.
